src/com/amazon/ion/BufferConfiguration.java

```diff
@@ -1,42 +1,39 @@
-package com.amazon.ion;
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
 
-import com.amazon.ion.impl.ReaderLookaheadBuffer;
+package com.amazon.ion;
 
 /**
  * Provides logic common to all BufferConfiguration implementations.
- * @param <Configuration> the type of the concrete subclass of this BufferConfiguration that is applicable to the
- *                        ReaderLookaheadBufferBase subclass.
+ * @param <Configuration> the type of the concrete subclass of this BufferConfiguration.
  */
 public abstract class BufferConfiguration<Configuration extends BufferConfiguration<Configuration>> {
```

Comment:
Curious what BufferConfigurations other than `IonBufferConfiguration` we are looking to support here.